08_voting/index.tex

```diff
@@ -1 +1,101 @@
+
+\section{Voting Protocol}
+\label{sec:VotingProtocol}
+The suggested voting protocol is adapted from an example of formal specification given in Robert's Rules of Order (RONR) that defines a series of sequential actions when a voting event happens.
+
+This protocol is applicable to any voting method at a certain state of the game.
+
+The sequence of actions of the voting protocol in this game when a voting event happens is as follows: The President selects a rule from the rule proposal list to be voted on; the President notifies the Speaker to hold a voting event for the selected rule; the Speaker calls for a vote for the selected rule by opening ballots to all islands to cast their votes; the Speaker closes the ballots after receiving votes from all islands; the Speaker counts the votes by calling an applicable voting method function for winner selection to produce a result; finally, the Speaker publicly declares the voting result whether the selected rule can be carried out or not. Figure~\ref{fig:RONRVotingProtocol} shows the sequence of the voting protocol for changing and/or adapting a new rule.
+
+
+\begin{figure}[H]
+\begin{center}
+\begin{tikzpicture}[node distance=20pt]
+\centering
+\node[draw, rounded corners] (start)  {The President selects a rule to be voted on};
+\node[draw, below=of start] (step 1)  {The President notifies the Speaker to hold a voting session on the selected rule};
+\node[draw, below=of step 1] (step 2)  {The Speaker opens the ballot for the Voters to cast their vote.};
+\node[draw, below=of step 2] (step 3)  {The Speaker closes the ballot after receiving votes from the Voters.};
+\node[draw, below=of step 3] (step 4)  {The Speaker counts the votes by using a specific voting method to declare the result.};
+\node[draw, below=of step 4, rounded corners] (end)   {The Speaker publicly declares the result for the finalised vote.};
+ \draw[->] (start)  -- (step 1);
+ \draw[->] (step 1) -- (step 2);
+ \draw[->] (step 2) -- (step 3);
+ \draw[->] (step 3) -- (step 4);
+ \draw[->] (step 4) -- (end);
+\end{tikzpicture}
+\caption{Voting Protocol for Rules}
+\label{fig:RONRVotingProtocol}
+\end{center}
+\end{figure}
+
+For elections of roles, the sequence of actions of the voting protocol is mostly similar to the above explanation in principle, except for some parameters, such as the motion of the vote which is the role itself (President, or Speaker, or Judge), the facilitator of the election/vote event which depends on what role is being held for election (refer to Chapter 5 IIGO for more details on change of roles and power transfer), and the applicable voting method function to call for election that will produce the result, which is different from the voting method used for rules. Refer to Section~\ref{sec:VotingMethods} for more details on voting methods to be used for elections of roles.
```

Comment:
this should be a reference to the chapter using the latex command - if the chapters are reordered and we miss something like this it could get really confusing and missed before submission

Reply:
so... what is wrong with this `ref{}` I don't understand your comment @yp717

Reply:
there wasn't a ref before he already changed it I believe

Reply:
nothing is wrong with that ref its perfect